src/app/provider/index.tsx

```diff
@@ -14,9 +16,17 @@ export default function Provider({
 }) {
   const queryClient = new QueryClient();
 
+   // put in context provider location
+  const [currentUser, setCurrentUser] = React.useState<CurrentUserContextObjectType | null>(null);
```

Comment:
This should stay within the current user context itfself for separate of concern. The index provider should just be all the providers imported wrapping one children. If we do it for every context it would just build up into a bunch of states within one file.